File: php/testcode/benchmark_test_00214.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00214(BenchmarkRequest $req): BenchmarkResponse {
    $pdo = getDbConnection();
    $status = $req->param('status');
    $raw = "status = '" . $status . "'";
    $result = $pdo->query("SELECT * FROM orders WHERE " . $raw);
    $rows = $result->fetchAll();
    return BenchmarkResponse::json($rows);
}

File: rust/testcode/benchmark_test_00214.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let body = req.param("body");
    let result = strict_deserialize(&body);
    match result {
        Ok(r) => super::shared::BenchmarkResponse::ok(&r),
        Err(e) => super::shared::BenchmarkResponse::bad_request(&e),
    }
}

fn strict_deserialize(_json: &str) -> Result<String, String> {
    Ok("StrictRequest { name: String }".to_string())
}
